geomstats/geometry/discrete_curves.py

```diff
@@ -1250,41 +1155,40 @@ class SRVMetric(PullbackDiffeoMetric):
 
     def __init__(
         self,
-        ambient_manifold,
-        ambient_metric=None,
+        space,
         translation_invariant=True,
-        k_sampling_points=10,
     ):
-        self.ambient_manifold = ambient_manifold
-        super().__init__(
-            dim=ambient_manifold.dim,
-            shape=(k_sampling_points,) + ambient_manifold.shape,
-        )
-        self.ambient_metric = ambient_metric
-        if ambient_metric is None:
-            if hasattr(ambient_manifold, "metric"):
-                self.ambient_metric = ambient_manifold.metric
-            else:
-                raise ValueError(
-                    "Instantiating an object of class "
-                    "ElasticMetric requires either a metric"
-                    " or an ambient manifold"
-                    " equipped with a metric."
-                )
+        super().__init__(space=space)
         self.translation_invariant = translation_invariant
 
-    def define_embedding_metric(self):
-        r"""Define the metric which is pulled back.
+    def _check_ambient_manifold(self, ambient_manifold):
+        if not isinstance(ambient_manifold, Euclidean):
+            raise AssertionError(
+                "This metric is only "
+                "implemented for discrete curves embedded "
+                "in a Euclidean space."
+            )
+
+    def define_embedding_space(self):
```

Comment:
I'll make it private, because in fact your example is something that should not be possible. With it private, it becomes closer to `LevelSet` and in fact fulfills the same goal.